GradeCenter.Data/Models/School.cs

```diff
@@ -0,0 +1,28 @@
+﻿using GradeCenter.Data.Models.Common;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace GradeCenter.Data.Models
+{
+    public class School: BaseModel<int>
+    {
+        public School()
```

Comment:
I think you can delete this constructor as you should initialize the collections directly.